overlord/auth/auth.go

```diff
@@ -183,3 +183,41 @@ NextUser:
 	}
 	return nil, ErrInvalidAuth
 }
+
+// An AuthContext handles user updates.
+type AuthContext interface {
+	UpdateUser(user *UserState) error
+}
+
+// AuthStateContext helps keeping track and updating users in the state.
+type AuthStateContext struct {
+	state *state.State
+}
+
+// NewAuthContext returns an AuthContext for state.
+func NewAuthStateContext(st *state.State) *AuthStateContext {
+	return &AuthStateContext{state: st}
+}
+
+// UpdateUser updates user in state.
+func (asc *AuthStateContext) UpdateUser(user *UserState) error {
+	var authStateData AuthState
+
+	asc.state.Lock()
+	defer asc.state.Unlock()
+
+	err := asc.state.Get("auth", &authStateData)
```

Comment:
Let's please move this logic to the `auth.UpdateUser(st, user)` function, and call it from here. It's a good partner to existing logic (NewUser, User, UpdateUser, RemoveUser).